Lib/test/test_capi/test_list.py

```diff
@@ -0,0 +1,85 @@
+import unittest
+import sys
+from test.support import import_helper
+from collections import UserList
+
+_testcapi = import_helper.import_module('_testcapi')
+
+NULL = None
+
+class ListSubclass(list):
+    ...
+
+
+class CAPITest(unittest.TestCase):
+    def test_check(self):
+        # Test PyList_Check()
+        check = _testcapi.list_check
+        self.assertTrue(check([1, 2]))
+        self.assertTrue(check([]))
+        self.assertTrue(check(ListSubclass([1, 2])))
+        self.assertFalse(check({1: 2}))
+        self.assertFalse(check((1, 2)))
+        self.assertFalse(check(42))
+        self.assertFalse(check(object()))
```

Comment:
Does it crash with NULL? If yes, please add the corresponding comment. All functions that take pointer arguments should be tested with NULL.